src/general/Jukebox.cs

```diff
@@ -38,6 +38,8 @@ public class Jukebox : Node
     /// </summary>
     private MusicCategory? previouslyPlayedCategory;
 
+    private MusicContext[] activeContexts = null!;
```

Comment:
I think this should be modified to allow null values, and null would be the same as general. That will improve a lot of the code here to not have to allocate dummy lists of data all over the place.